File: android-app/app/src/main/java/com/example/mobileapp/models/Favorite.kt
package com.example.mobileapp.models

data class Favorite(
    val id: Int,
    val favoriteId: Int,
    val userId: String,
    val productId: Int,
    val createdAt: String
)

data class FavoriteRequest(
    val productId: Int
)

data class FavoriteResponse(
    val data: Favorite,
    val timestamp: String
)

data class FavoriteListResponse(
    val data: List<Favorite>,
    val timestamp: String
)

File: ios-app/MobileApp/MobileApp/Models/Favorite.swift
//
//  Favorite.swift
//  MobileApp
//
//  Created on 2025-01-08.
//

import Foundation

struct Favorite: Codable, Identifiable {
    let id: Int
    let favoriteId: Int
    let userId: String
    let productId: Int
    let createdAt: String
}

struct FavoriteRequest: Codable {
    let productId: Int
}

struct FavoriteResponse: Codable {
    let data: Favorite
    let timestamp: String
}

struct FavoriteListResponse: Codable {
    let data: [Favorite]
    let timestamp: String
}
